storage/postgres/admin_storage.go

```diff
@@ -278,19 +343,112 @@ func (t *adminTX) CreateTree(ctx context.Context, tree *trillian.Tree) (*trillia
 }
 
 func (t *adminTX) UpdateTree(ctx context.Context, treeID int64, updateFunc func(*trillian.Tree)) (*trillian.Tree, error) {
-	return nil, errNotImplemented
+	tree, err := t.GetTree(ctx, treeID)
+	if err != nil {
+		return nil, err
+	}
+
+	beforeUpdate := *tree
+	updateFunc(tree)
+	if err := storage.ValidateTreeForUpdate(ctx, &beforeUpdate, tree); err != nil {
+		return nil, err
+	}
+	if err := validateStorageSettings(tree); err != nil {
+		return nil, err
+	}
+
+	// Use the time truncated-to-millis throughout, as that's what's stored.
+	nowMillis := storage.ToMillisSinceEpoch(time.Now())
+	now := storage.FromMillisSinceEpoch(nowMillis)
+	tree.UpdateTime, err = ptypes.TimestampProto(now)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build update time: %v", err)
```

Comment:
s/update time/tree.UpdateTime/

Makes it a bit easier to recognise, from the error message, what it was trying to do.